numpy/testing/_private/pytesttester.py

```diff
@@ -0,0 +1,329 @@
+
+        # Set to "release" in constructor in maintenance branches.
+        self.raise_warnings = raise_warnings
+
+        # Whether to check for FPU mode changes
+        self.check_fpu_mode = check_fpu_mode
+
+    def test(self, label='fast', verbose=1, extra_argv=None,
+             doctests=False, coverage=False, raise_warnings=None,
+             timer=False, tests=None):
+        """
+        Run tests for module using nose.
+
+        Parameters
+        ----------
+        label : {'fast', 'full'}, optional
+            Identifies the tests to run. When set to 'fast', tests decorated
+            with `pytest.mark.slow` are skipped, when 'full', the slow marker
+            is ignored.
+        verbose : int, optional
+            Verbosity value for test outputs, in the range 1-3. Default is 1.
+        extra_argv : list, optional
+            List with any extra arguments to pass to pytests.
+        doctests : bool, optional
+            .. note:: Not supported
+        coverage : bool, optional
+            If True, report coverage of NumPy code. Default is False.
+            Requires installation of (pip) pytest-cov.
+        raise_warnings : None, str or sequence of warnings, optional
+            This specifies which warnings to configure as 'raise' instead
+            of being shown once during the test execution.  Valid strings are:
+
+            * "develop" : equals ``(Warning,)``
+            * "release" : equals ``()``, don't raise on any warnings.
+
+            The default is to use the class initialization value.
+        timer : bool or int, optional
+            If not None, report the time of the slowest `timer` tests.
+
+        tests : test or list of tests
+            Tests to be executed with pytest '--pyargs'
+
+        Returns
+        -------
+        result : bool
+            Return True on success, false otherwise.
+
+        Notes
+        -----
+        Each NumPy module exposes `test` in its namespace to run all tests for it.
+        For example, to run all tests for numpy.lib:
+
+        >>> np.lib.test() #doctest: +SKIP
+
+        Examples
+        --------
+        >>> result = np.lib.test() #doctest: +SKIP
+        Running unit tests for numpy.lib
+        ...
+        Ran 976 tests in 3.933s
+
+        OK
+
+        >>> result.errors #doctest: +SKIP
+        []
+        >>> result.knownfail #doctest: +SKIP
+        []
+
+        """
+        import pytest
+
+        # cap verbosity at 3, which is equivalent to the pytest '-vv' option
+        verbose = min(int(verbose), 3)
+
+        #FIXME This is no longer needed? Assume it was for use in tests.
+        from . import utils
+        utils.verbose = verbose
+
+        # setup the pytest arguments
+        pytest_args = ['-l']
+
+        if doctests:
+            raise ValueError("Doctests not supported")
+
+        if extra_argv:
+            pytest_args += list(extra_argv)
+
+        if verbose and int(verbose) > 1:
```

Comment:
Same remark about casting `verbose` - why not just require an integer to be passed, rather than allowing floats and strings?

Reply:
If that's the intent, `operator.index` is a better way to spell that. I'd probably leave it without the cast though